### PR TITLE
Merge branch '1.3' of ezsystems/ezplatform-rest into 4.4

### PR DESCRIPTION
**Pull request by ezrobot**
Cross merge PR: https://github.com/ezsystems/ezplatform-rest/pull/102

### CODE REVIEW

File: tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location as ApiLocation;
+use eZ\Publish\Core\Base\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\Core\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentType;
+use EzSystems\EzPlatformRest\Server\Output\ValueObjectVisitor;
+use EzSystems\EzPlatformRest\Tests\Output\ValueObjectVisitorBaseTest;
```
Comment (alongosz) [inline tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php:20]: Legacy namespaces @Steveb-p @barw4

File: tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    private const UNAUTHORIZED_MAIN_LOCATION_ID = 111;
+
+    private const LOCATION_ID = 55;
+
+    /** @var \eZ\Publish\API\Repository\LocationService|\PHPUnit\Framework\MockObject\MockObject */
+    private $locationServiceMock;
+
+    /** @var \eZ\Publish\API\Repository\ContentService|\PHPUnit\Framework\MockObject\MockObject */
+    private $contentServiceMock;
```
Comment (alongosz) [inline tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php:34]: Here as well. Btw while this doc needs to stay, I think we can use here strong type hint maybe hinting the actual objects - LocationService and ContentService, like that:
```suggestion
    /** @var \Ibexa\Contracts\Core\Repository\LocationService&\PHPUnit\Framework\MockObject\MockObject */
    private LocationService $locationServiceMock;

    /** @var \Ibexa\Contracts\Core\Repository\ContentService&\PHPUnit\Framework\MockObject\MockObject */
    private ContentService $contentServiceMock;
```

WDYT?

**Review by alongosz — CHANGES_REQUESTED** [review]

File: tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php (hunk 2 above)
Comment (barw4) [reply tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php:34]: I think it was discussed is some other PR that we won't be strong-type hinting yet.

**Review by alongosz — APPROVED** [review]

File: tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php (hunk 2 above)
Comment (alongosz) [reply tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php]: > I think it was discussed is some other PR that we won't be strong-type hinting yet.

AFAIK the discussion you're referring to was rather about whether we should obfuscate the diff with superfluous changes to the unrelated lines of code, especially when handling cross-merges. If it's justified, it depends on a specific diff.
Not type-hinting is a technical debt.

I prefer to have it when possible, so I amended merge commit with the change.

Comment (barw4) [reply tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php]: This was another discussion, I referred to https://github.com/ibexa/fieldtype-page/pull/66#pullrequestreview-1299275967. 

I discussed this with Konrad at that time and seems like we aren't really using those, if the consensus is to use them I'm good with it.

**Pull request merged**